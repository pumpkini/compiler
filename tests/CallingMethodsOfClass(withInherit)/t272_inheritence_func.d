class Base {
    int a;
    void f(int x) {
        int a;
        this.a = x;
        a = 10;
        Print(a);
    }
    void print(){
        Print(this.a);
        Print(a);
    }
    void g(int x) {
        Print("HEELO");
    }
}

class Derived extends Base{
    int b;
    void g(int y) {
        this.print();
        Print(y);
    }
}

int main() {
    Base b;
    Derived d;
    d = new Derived;
    d.f(100);
    d.g(200);
    b = d;
    b.f(100);
    b.g(400);
    b = new Base;
    b.g(400);
}
