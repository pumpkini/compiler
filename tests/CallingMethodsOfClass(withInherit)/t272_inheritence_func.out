10
100
100
200
10
100
100
400
HEELO
